### ISSUE TITLE
Astro - Cannot find module '@vercel/speed-insights/astro' on import

### ISSUE THREAD

**Issue by ref-thomasledoux1** [vercel/speed-insights#36]
When I try to import the Astro package, I get a TS error:

Cannot find module '@vercel/speed-insights/astro' or its corresponding type declarations.ts(2307)

Code in .astro file: 
```import SpeedInsights from "@vercel/speed-insights/astro";```

File can be found here:
https://github.com/thomasledoux1/website-thomas-astro/blob/main/src/layouts/Layout.astro

**Comment by alexbouchardd** [comment]
Same here. Package is seemingly not exported.

**Comment by feugy** [comment]
Hello gents and thank you for this bug report.

I did not managed to reproduce the error.
I've setup a basic Astro application:
```shell
> pnpm create astro@latest

 astro   Launch sequence initiated.

   dir   Where should we create your new project?
         ./speed-insights-issue-36

  tmpl   How would you like to start your new project?
         Include sample files

  deps   Install dependencies?
         Yes

    ts   Do you plan to write TypeScript?
         No
      ◼  No worries! TypeScript is supported in Astro by default,
         but you are free to continue writing JavaScript instead.

   git   Initialize a new git repository?
         No
      ◼  Sounds good! You can always run git init manually.

      ✔  Project initialized!
         ■ Template copied
         ■ Dependencies installed

  next   Liftoff confirmed. Explore your project!

         Enter your project directory using cd ./speed-insights-issue-36
         Run pnpm dev to start the dev server. CTRL+C to stop.
         Add frameworks like react or tailwind using astro add.

         Stuck? Join us at https://astro.build/chat

╭─────╮  Houston:
│ ◠ ◡ ◠  Good luck out there, astronaut! 🚀
╰─────╯

> cd speed-insights-issue-36

> pnpm i @vercel/speed-insights
Packages: +1
+
Progress: resolved 450, reused 416, downloaded 0, added 1, done

dependencies:
+ @vercel/speed-insights 1.0.2

Done in 906ms
```

Then in `src/layouts/Layout.astro` I added:
```astro
---
import SpeedInsights from '@vercel/speed-insights/astro'; // <---- added

interface Props {
  title: string;
}

const { title } = Astro.props;
---

<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="description" content="Astro description" />
    <meta name="viewport" content="width=device-width" />
    <link rel="icon" type="image/svg+xml" href="/favicon.svg" />
    <meta name="generator" content={Astro.generator} />
    <title>{title}</title>
  </head>
  <body>
    <slot />
    <SpeedInsights />  <!-- <---- added -->
  </body>
</html>
```

Finally, running `pnpm dev` and visiting the site works seamlessly.

@alexbouchardd the astro component export is properly defined [in  packages/web/package.json](https://github.com/vercel/speed-insights/blob/main/packages/web/package.json#L22).

The types are not, indeed, but we're considering replacing Tsup, so we'll get that fix during the migration.

**Comment by mingjunlu** [comment]
Hi @feugy,

I've created a [minimal reproduction repo](https://github.com/mingjunlu/speed-insights-issue-36).

Steps to reproduce:

```bash
git clone git@github.com:mingjunlu/speed-insights-issue-36.git
cd speed-insights-issue-36
pnpm install
pnpm run build
```

You'll see some error message like this:

```
src/pages/index.astro:2:27 - error ts(2307): Cannot find module '@vercel/speed-insights/astro' or its corresponding type declarations.

2 import SpeedInsights from '@vercel/speed-insights/astro';
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Result (3 files): 
- 1 error
- 0 warnings
- 0 hints

 ELIFECYCLE  Command failed with exit code 1.
```

**Comment by feugy** [comment]
We've released version 1.0.4 which fixes the issue. Happy coding!
